components/browser/icons/src/main/java/mozilla/components/browser/icons/processor/PersistingIconProcessor.kt

```diff
@@ -0,0 +1,12 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.icons.processor
+
+import mozilla.components.browser.icons.Icon
+
+/**
+ * [IconProcessor] variation that will persist the received [Icon].
+ */
+interface PersistingIconProcessor : IconProcessor
```

Comment:
Right now all of the default processors are persisting processors, right? If there are other processors then we may still want to run them, I think - since we may want to cache the processed image. 🤔

Reply:
Indeed, thank you! Seems like this marker interface is not needed.
Though for the future refactoring I'd still find it useful to have a separation between persistence processors and others just to have a way to enforce the order between these. 
A DSL to make it easier for clients to set the appropriate processors (or other params)  and for BrowserIcons to enforce the order and scopes would also be nice (was thinking now that for this example even each processor could be scoped and have a "then" method accepting other processors to easily transmit to BrowserIcons clients that the order is important).